Return dashboard in GET "/most_recently_viewed_dashboard" response

calherries opened the pull request:
Pursuant to https://github.com/metabase/metabase/issues/30140
Epic: https://github.com/metabase/metabase/issues/29666

Builds on https://github.com/metabase/metabase/pull/30323#top

This PR adds an estension to the previous PR, namely returning a dashboard object in the response instead of a dashboard id.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30566)
<!-- Reviewable:end -->

Commit:
Only update dashboard if changes have happened

Commit:
Add GET "/most_recently_viewed_dashboard" endpoint

Commit:
Add endpoint

Commit:
Undo accidentally committed changes to api/dashboard/:id

Commit:
Undo accidental changes to record-view!

Commit:
Fix test

Commit:
Fix test docstring

Commit:
Return dashboard object and add permissions check

calherries committed:
Merge branch 'question-to-dash-flow/main' into 30140-backend-most-recently-viewed-dashboard-extension

dpsutton reviewed (approved):
I'm approving but i'm not sure about the semantics of returning a 404 if the route doesn't have a recent dashboard. It feels off to me. The 204 felt more natural but up to yall

uladzimirdev commented:
@dpsutton right, I'm for 204 as well

john-metabase committed:
Fixes IP geocoding test (#30605)

dpsutton committed:
Return 204 if no recent viewed dashboard

Three cases to handle:
- no recent dashboard
- recent dashboard but lack permissions
- recent dashboard you can see

the FE really just cares "is there a recent dashboard to show you" which
logically collapses the first two cases. So we return a generic 204
to the FE. In the third case we return the dashboard.

uladzimirdev committed:
Typo

uladzimirdev committed:
Merge remote-tracking branch 'origin/question-to-dash-flow/main' into 30140-backend-most-recently-viewed-dashboard-extension
